09_class_bank_card.py

```diff
@@ -0,0 +1,261 @@
+"""Home task. Bank card operations.
+
+write class for bank card.
+Class should reflect card lifecycle, card operations etc.
+use CVV, PIN, Name, Surname , end date, card_num as initial params
+class should have in addition to common logic some class attributes,
+as minimum one classmethod and as minimum  one staticmethod,
+two or more getters/setters, __str_ magic method
+do not forget about block "if __name__ == '__main__':"
+and code there to check your class written logic.
+"""
+
+import datetime
+import random
+
+
+class Creditcard:
+    """The class represent of Credit cards."""
+
+    CURRENT_DATE = datetime.date.today()
+    PIN_LENGTH = 4
+
+    def __init__(self, pin, name, surname):
+        """Initialize an instance of the Credit card.
+
+        param pin: An integer representing the pin code.
+        param name: A string representing the name of the client.
+        param surname: A string representing the surname of the client.
+        param cvv: Automatic generation of CVV code in integer format.
+        param exp_date: Automatically create a date from today + 5 years.
+        param card_num: Automatic generation of card number in integer format.
+        param balance: An integer representing the balance on card.
+        """
+        self.__pin = pin
+        self.name = name
+        self.surname = surname
+        self.__cvv = random.randint(100, 999)
+        self.__exp_date = datetime.date.today().replace(
```

Comment:
self.__exp_date = datetime.date.today().replace(year=datetime.date.today().year + 5, day=1)  More safe way cause different months have different number of days ))
For example Today Feb 29 that not exists in 2029 )